pkg/serviceaccount/tokens_controller.go

```diff
@@ -240,8 +244,16 @@ func (e *TokensController) secretDeleted(obj interface{}) {
 		return
 	}
 
-	if _, err := e.removeSecretReferenceIfNeeded(serviceAccount, secret.Name); err != nil {
-		glog.Error(err)
+	for i := 1; i <= NumServiceAccountRemoveReferenceRetries; i++ {
```

Comment:
In that case, is it important to remove non-existing secrets from this list?

I feel like the water is very muddy here, in terms of what objects own/depend on other objects. Maybe after 1.0 we can try to simplify this; it's difficult to reason about at the moment.